File: VHBoomMenuButtonDemo/BlurBackgroundAndTipVC.m
//
//  BlurBackgroundAndTipVC.m
//  VHBoomMenuButtonDemo
//
//  Created by Nightonke on 2017/4/21.
//  Copyright © 2017 Nightonke. All rights reserved.
//

#import "BlurBackgroundAndTipVC.h"
#import "BuilderManager.h"
#import <BoomMenuButton/BoomMenuButton.h>

@interface BlurBackgroundAndTipVC ()

@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb1;
@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb2;
@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb3;

@end

@implementation BlurBackgroundAndTipVC

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.bmb1.buttonEnum = VHButtonSimpleCircle;
    self.bmb1.piecePlaceEnum = VHPiecePlaceDOT_9_1;
    self.bmb1.buttonPlaceEnum = VHButtonPlaceSC_9_1;
    self.bmb1.pieceCornerRadius = 1;
    self.bmb1.backgroundBlurred = YES;
    self.bmb1.blurEffect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleExtraLight];
    self.bmb1.tip = @"Select an animal to feed!";
    self.bmb1.tipBelowButtons = YES;
    self.bmb1.tipButtonMargin = 30;
    self.bmb1.tipLabel.textColor = [UIColor darkGrayColor];
    for (int i = 0; i < self.bmb1.pieceNumber; i++)
    {
        [self.bmb1 addBuilder:[BuilderManager squareSimpleCircleButtonBuilder]];
    }
    
    self.bmb2.buttonEnum = VHButtonHam;
    self.bmb2.piecePlaceEnum = VHPiecePlaceHAM_4;
    self.bmb2.buttonPlaceEnum = VHButtonPlaceHAM_4;
    self.bmb2.backgroundBlurred = YES;
    self.bmb2.blurEffect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleDark];
    self.bmb2.tip = @"Nightonke BMB";
    for (int i = 0; i < [VHBoomMenuButton pieceNumber:self.bmb2.piecePlaceEnum]; i++)
    {
        [self.bmb2 addBuilder:[BuilderManager hamButtonBuilderWithDifferentPieceColor]];
    }
    
    self.bmb3.buttonEnum = VHButtonTextOutsideCircle;
    self.bmb3.piecePlaceEnum = VHPiecePlaceDOT_9_1;
    self.bmb3.buttonPlaceEnum = VHButtonPlaceSC_9_1;
    self.bmb3.backgroundBlurred = YES;
    self.bmb3.blurEffect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleLight];
    self.bmb3.tip = @"Select an animal to feed!";
    self.bmb3.tipLabel.textColor = [UIColor darkGrayColor];
    self.bmb3.tipLabel.font = [UIFont systemFontOfSize:15];
    for (int i = 0; i < [VHBoomMenuButton pieceNumber:self.bmb3.piecePlaceEnum]; i++)
    {
        VHTextOutsideCircleButtonBuilder *builder = [BuilderManager textOutsideCircleButtonBuilder];
        builder.normalTextColor = [UIColor darkGrayColor];
        builder.highlightedTextColor = [UIColor lightGrayColor];
        [self.bmb3 addBuilder:builder];
    }
}

@end

File: VHBoomMenuButtonSwiftDemo/BlurBackgroundAndTipVC.swift
//
//  BlurBackgroundAndTipVC.swift
//  VHBoomMenuButtonSwiftDemo
//
//  Created by viktorhuang on 2017/5/5.
//  Copyright © 2017年 Nightonke. All rights reserved.
//

import UIKit
import BoomMenuButton

class BlurBackgroundAndTipVC: BoomMenuButtonVC {

    @IBOutlet weak var bmb1: BoomMenuButton!
    @IBOutlet weak var bmb2: BoomMenuButton!
    @IBOutlet weak var bmb3: BoomMenuButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        bmb1.buttonEnum = .simpleCircle
        bmb1.piecePlaceEnum = .dot_9_1
        bmb1.buttonPlaceEnum = .sc_9_1
        bmb1.pieceCornerRadius = 1
        bmb1.backgroundBlurred = true
        bmb1.blurEffect = UIBlurEffect.init(style: UIBlurEffectStyle.extraLight)
        bmb1.tip = "Select an animal to feed!"
        bmb1.tipButtonMargin = 30
        bmb1.tipLabel().textColor = UIColor.darkGray
        for _ in 0..<bmb1.piecePlaceEnum.pieceNumber() {
            bmb1.addBuilder(BuilderManager.squareSimpleCircleButtonBuilder())
        }
        
        bmb2.buttonEnum = .ham
        bmb2.piecePlaceEnum = .ham_4
        bmb2.buttonPlaceEnum = .ham_4
        bmb2.backgroundBlurred = true
        bmb2.blurEffect = UIBlurEffect.init(style: UIBlurEffectStyle.dark)
        bmb2.tip = "Nightonke BMB"
        for _ in 0..<bmb2.piecePlaceEnum.pieceNumber() {
            bmb2.addBuilder(BuilderManager.hamButtonBuilderWithDifferentPieceColor())
        }
        
        bmb3.buttonEnum = .textOutsideCircle
        bmb3.piecePlaceEnum = .dot_9_1
        bmb3.buttonPlaceEnum = .sc_9_1
        bmb3.backgroundBlurred = true
        bmb3.blurEffect = UIBlurEffect.init(style: UIBlurEffectStyle.light)
        bmb3.tip = "Select an animal to feed"
        bmb3.tipLabel().textColor = UIColor.darkGray
        bmb3.tipLabel().font = UIFont.systemFont(ofSize: 15)
        for _ in 0..<bmb3.piecePlaceEnum.pieceNumber() {
            let builder = BuilderManager.textOutsideCircleButtonBuilder()
            builder.normalTextColor = UIColor.darkGray
            builder.highlightedTextColor = UIColor.lightGray
            bmb3.addBuilder(builder)
        }
    }
}
